script.js

```diff
@@ -1,4 +1,91 @@
+//global variables
+var currentGameMode = `waiting for username`;
+var userName = ``;
+//Number Games of Won
+var numGamesPlayed = 0;
+console.log(numGamesPlayed);
+//Number of Games Played
+var numGamesWon = 0;
+console.log(numGamesWon);
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = ``;
+  if (currentGameMode == `waiting for username`) {
+    //set the name
+    userName = input;
+    console.log(userName);
+    currentGameMode = `SPSGame`;
+
+    myOutputValue = `Hello ${userName}`;
+  } else if (currentGameMode == `SPSGame`) {
+    //Increase number of Games Played
+    numGamesPlayed += 1;
+
+    //SPS Generator
+    var SPSArray = ["Scissors", "Paper", "Stone"];
+    var randomSPS = Math.floor(Math.random() * SPSArray.length); //RandomSPS
```

Comment:
as this variable doesn't give us scissors paper or stone yet, randomInt may be a better name